crates/virtio-device/src/mmio.rs

```diff
@@ -35,11 +35,14 @@ fn update_queue_field<M, D, F>(device: &mut D, f: F)
 where
     M: GuestAddressSpace,
     D: WithDriverSelect<M> + ?Sized,
-    F: FnOnce(&mut Queue<M>),
+    F: FnOnce(&mut Queue<M>) -> Result<(), Error>,
 {
     if device.check_device_status(status::FEATURES_OK, status::DRIVER_OK | status::FAILED) {
         if let Some(queue) = device.selected_queue_mut() {
-            f(queue);
+            if let Err(e) = f(queue) {
+                device.set_device_status(status::FAILED);
```

Comment:
From the experiments that I did last week, setting the device status to FAILED doesn't seem to actually make a difference. Sorry for not coming with a response on the issue, but I had a chat only today with @andreeaflorescu and @alexandruag to decide on the path forward here, and we were thinking to keep the setter proposal, but in case of a misconfiguration to set the value of that field to a default. This seems to be how qemu handles this issue as well. So for the queue size, we would keep the max size if the value is invalid, and for the descriptor table address and the others, we would use address 0. What do you think?
I think with this approach, we could get rid of the `actual_size()` method as well, but I'll double-check.